projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts

```diff
@@ -0,0 +1,150 @@
+import {AfterViewInit, Component, ViewChild} from '@angular/core';
+import {
+    TaskEventNotification,
+    TaskViewService,
+    CaseResourceService,
+    PublicCaseResourceService,
+    PublicTaskResourceService,
+    SearchService,
+    PublicProcessService,
+    ProcessService,
+    TaskResourceService,
+    SnackBarService,
+    UserService,
+    SessionService,
+    PetriNetResourceService,
+    PublicPetriNetResourceService,
+    LoggerService,
+    ResourceProvider,
+    ConfigurationService,
+    FieldConverterService,
+    AuthenticationService,
+    PublicUrlResolverService,
+    publicBaseFilterFactory,
+    publicFactoryResolver,
+    AllowedNetsService,
+    PublicTaskLoadingService,
+    AllowedNetsServiceFactory,
+    NAE_VIEW_ID_SEGMENT,
+    ViewIdService,
+    NAE_BASE_FILTER,
+    ChangedFieldsService,
+    AbstractSingleTaskViewComponent,
+    TaskContentService,
+    TaskDataService,
+    FinishTaskService,
+    SingleTaskContentService, TaskRequestStateService, TaskEventService, NAE_TASK_OPERATIONS, SubjectTaskOperations
+} from '@netgrif/components-core';
+import {HeaderComponent} from '@netgrif/components';
+import {ActivatedRoute, Router} from '@angular/router';
+import {TranslateService} from '@ngx-translate/core';
+import {combineLatest} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+const localAllowedNetsServiceFactory = (factory: AllowedNetsServiceFactory, route: ActivatedRoute) => {
+    const array = [];
+    if (route.snapshot.paramMap.get('petriNetId') !== null) {
+        array.push(route.snapshot.paramMap.get('petriNetId'));
+    }
+    return factory.createFromArray(array);
+};
+
+const processServiceFactory = (userService: UserService, sessionService: SessionService, authService: AuthenticationService,
```

Comment:
I think this could be done in standalone task, as all of the example process, task and case views contain the same implementation, so it would result in bigger refactor of components.

Reply:
I don't really care what the example app uses. But what the developers use in their projects. But I agree that if the change is large it should be done separately.